docs/CPS/CPS-XXXX/README.md

```diff
@@ -0,0 +1,113 @@
+- This complicates or inhibits products offering a full node experience, with a free and open source service, available for users to install then choose their wallet of choice.
+
+## Use cases
+
+### Lace Extension
+
+#### What is it doing?
+- **Long-polling the tip for changes**. An HTTP GET request is made to a webserver every 5 seconds, checking if there's a new tip in the [Cardano DB Sync] database. If so, requests are then made to fetch transactions that are associated with the provided list of addresses.
+- Transaction data in responses are **transmitted as JSON**, via a **custom schema**
+- Data is sourced from the combination of [Cardano DB Sync], Ouroboros Local State Queries, and custom Ouroboros Chain-Sync projections into the PostgreSQL DB.
+
+> [!NOTE]
+> Requests are also made to fetch data on-demand but the scope of this document is to focus on the fundamental wallet requirements, so it's not mentioned here.
+
+<img alt="Lace fundamental state sequence flow diagram" src="images/Lace_Fundamental_State_Sequence_Flow.png">
+
+#### Why?
+- The philosophy in Lace is to place the responsibility of deriving wallet state from the historical transactions, similar to projecting state via the chain-sync protocol.
+- A custom schema was implemented due to the lack of a standard
+- 
+> [!NOTE]
+> Since this API was developed, [Cardano DB Sync] has since added a mode to store transactions in CBOR, but at the time of implementation it was not an option.
+
+#### Current Alternatives
+[Blockfrost WebSocket Link] is a partial solution for the impedance mismatch problem, however it's not addressing the fragility and standards problem. It's not practical to run this service to satisfy a single-tenant, full node, use case. 
+
+## Goals
+- Define a standard API that is optimal for fundamental wallet operation. Consider extensibility for optional enhancement.
+- Deliver data as a filtered subset of the network state, using native encoding formats, to simplify server implementations and push processing to the edge of the network.
+- Foster long term stability of the API when crossing major Cardano protocol upgrade boundaries.
+- Facilitate event-driven wallet behavior, minimising the need for long-polling, and to increase the sophistication of infrastructure management and user-base insights.
+- Support script-based wallets
+- Minimise the network demand by allowing the server to control the throttling strategy, rather than a pull-based approach like we see in the local chain-sync protocol. 
+- API can be applied in a local, single-tenant environment, and a remote multi-tenant deployment, to facilitate a universal approach to wallet supporting an optional full local node for power users. 
+- Remove the need for wallets to perform address discovery
+- Make it possible for a user to quickly derive new accounts and their addresses without re-projecting from genesis. 
```

Comment:
Subscribing with account pub key is a reasonable option, but it can only be used to derive payment/staking keys for that account - wouldn't be able to discover other accounts. Also for script wallets, we'd still need the credentials (or script hashes).

I believe account discovery also wouldn't work if we wanted to be subscribing with root pub key because of the 'hardening'. That is why we needed to add password prompt for enabling accounts when we implemented multi-account.

Reply:
Ah good point. I might just drop this out of the document, as again, it's more of a concern relating to a single-tenant  implementation.

Reply:
https://github.com/will-break-it/wallet-architecture/pull/3/commits/da24badbf4b5c0abf5f75f2323842829afef4fdd